### ISSUE TITLE
If-Modified-Since header not working with http2 connection

### ISSUE THREAD

**Issue by danhuynhdev** [ericmj/mint#171]
Instead of returning the body and status 304, it return error with reason `{:server_closed_request, :protocol_error}`.
```elixir
{:ok, conn} = Mint.HTTP2.connect(:https, "varvy.com", 443)
req_headers = [{"If-Modified-Since", "Wed, 26 May 2019 07:43:40 GMT"}]
{:ok, conn, request_ref} = Mint.HTTP2.request(conn, "GET", "/", req_headers)
next_message =
  receive do
    msg -> msg
  end
{:ok, conn, responses} = Mint.HTTP2.stream(conn, next_message)
next_message =
  receive do
    msg -> msg
  end
{:ok, conn, responses} = Mint.HTTP2.stream(conn, next_message)
# responses => {:error, _ref,
#   %Mint.HTTPError{module: Mint.HTTP2, reason: {:server_closed_request, :protocol_error}
# }}
next_message =
  receive do
    msg -> msg
  end
{:ok, conn, responses} = Mint.HTTP2.stream(conn, next_message)
# 18:38:55.849 [debug] Got frame: {:goaway, 0, 0, 0, :no_error, "timeout"}
```
Doing the same with Mint.HTTP1 and curl works.
```elixir
{:ok, conn} = Mint.HTTP1.connect(:https, "varvy.com", 443)
req_headers = [{"If-Modified-Since", "Wed, 26 May 2019 07:43:40 GMT"}]
{:ok, conn, request_ref} = Mint.HTTP1.request(conn, "GET", "/", req_headers)
next_message =
  receive do
    msg -> msg
  end
{:ok, conn, responses} = Mint.HTTP1.stream(conn, next_message)
# response => {:ok, ...}
```

Related ticked https://github.com/appcues/mojito/issues/24

**Comment by whatyouhide** [comment]
Interesting @danhuynhdev. We don't treat `If-Modified-Since` any different than any other headers. From your report, it seems like the server is closing the connection with a `GOAWAY` frame. What server are you using?

**Comment by optikfluffel** [comment]
@whatyouhide have a look at [appcues/mojito#24](https://github.com/appcues/mojito/issues/24) for more details

**Comment by whatyouhide** [comment]
You were right, it was a bug in Mint. Fixed in #172. 👍
